File: Frontend/app/src/app/services/alumnoGrupo/alumno-grupo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AlumnoResponse } from '../../models/alumno.model';

@Injectable({
  providedIn: 'root'
})
export class AlumnoGrupoService {


  private apiUrl = 'http://localhost:8080/api/alumno-grupo';

  constructor(
    private http: HttpClient
  ) { }

  asignarAlumno(grupoId: number, alumnoId: number): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/${grupoId}/alumnos/${alumnoId}`, {});
  } 

  quitarAlumno(grupoId: number, alumnoId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${grupoId}/alumnos/${alumnoId}`);
  }

  listarAlumnosDelGrupo(grupoId: number): Observable<AlumnoResponse[]> {
    return this.http.get<AlumnoResponse[]>(`${this.apiUrl}/${grupoId}/alumnos`);
  }

 

}

File: Backend/demo/src/main/java/com/example/demo/service/AlumnoGrupoService.java
package com.example.demo.service;

import com.example.demo.model.AlumnoGrupo;

import java.util.List;

public interface AlumnoGrupoService {

    AlumnoGrupo asignarAlumno(Long grupoId, Long alumnoId);

    List<AlumnoGrupo> listarAlumnosPorGrupo(Long grupoId);

    List<AlumnoGrupo> listarGruposPorAlumno(Long alumnoId);

    void quitarAlumno(Long grupoId, Long alumnoId);
}
